### PR TITLE
Fixes #8004 umbnodepreview setting contentType alias to null

### LINKED ISSUE #8004: Backoffice Users without access to "Settings" can't create new Members
We're experiencing an issue with Umbraco 8.6 and creating new members in the Backoffice UI. This only effects users that don't have access to the "Settings"-section and it also seems like the "debug=true/false"-setting in web.config has a role in this.

## Umbraco version:
I am seeing this issue on Umbraco version: 8.6.0

Reproduction
------------
Make sure that web.config has "debug=false" set so that JS-files are bundled and minified in the backoffice.

Create a new User that has access to the "Members"-section but does not have access to the "Settings"-section and login as this User.

Go to the members-section and "Create a member", fill out required fields and click save.

After this I'm given this error:

error message

It's really strange but if I switch back to "debug=true" in web.config this works. I've made sure to bump the ClientDependencyVersion number so it's not a cache-issue with the client files.

### Bug summary
As far as I can tell, the "contentTypeAlias" is returned from the "GetEmpty" call to the server when the "Create new Member"-page is loaded:

get

But when the payload is sent to the server to be saved, the alias is no longer there:

capture2

The server side error creates the following log entry:

```
{"@t":"2020-04-23T11:45:26.7603557Z","@mt":"Unhandled controller exception occurred for request '{RequestUrl}'","@l":"Error","@x":"System.InvalidOperationException: No member type found with alias \r\n   at Umbraco.Web.Editors.Binders.MemberBinder.CreateNew(MemberSave model) in D:\\a\\1\\s\\src\\Umbraco.Web\\Editors\\Binders\\MemberBinder.cs:line 135\r\n   at Umbraco.Web.Editors.Binders.MemberBinder.BindModel(HttpActionContext actionContext, ModelBindingContext bindingContext) in D:\\a\\1\\s\\src\\Umbraco.Web\\Editors\\Binders\\MemberBinder.cs:line 59\r\n   at System.Web.Http.ModelBinding.ModelBinderParameterBinding.ExecuteBindingAsync(ModelMetadataProvider metadataProvider, HttpActionContext actionContext, CancellationToken cancellationToken)\r\n   at System.Web.Http.Controllers.HttpActionBinding.<ExecuteBindingAsyncCore>d__12.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ActionFilterResult.<ExecuteAsync>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()","RequestUrl":"https://..../umbraco/backoffice/UmbracoApi/Member/PostSave","SourceContext":"Umbraco.Web.Editors.MemberController","ProcessId":12496,"ProcessName":"w3wp","ThreadId":65,"AppDomainId":20,"AppDomainAppId":"LMW3SVC76ROOT","MachineName":"OBVI-WEB","Log4NetLevel":"ERROR","HttpRequestNumber":16,"HttpRequestId":"fbf82615-4796-4e80-b882-f07270ca7d21","RequestData":"------WebKitFormBoundaryt0BjYAAZkFW5wK8D\r\nContent-Disposition: form-data; name=\"contentItem\"\r\n\r\n{\"id\":0,\"properties\":[{\"id\":0,\"alias\":\"umbracoMemberComments\",\"value\":\"\"},{\"id\":0,\"alias\":\"umbracoMemberFailedPasswordAttempts\",\"value\":\"\"},{\"id\":0,\"alias\":\"umbracoMemberApproved\",\"value\":\"1\"},{\"id\":0,\"alias\":\"umbracoMemberLockedOut\",\"value\":\"No\"},{\"id\":0,\"alias\":\"umbracoMemberLastLockoutDate\",\"value\":\"\"},{\"id\":0,\"alias\":\"umbracoMemberLastLogin\",\"value\":\"\"},{\"id\":0,\"alias\":\"umbracoMemberLastPasswordChangeDate\",\"value\":\"\"},{\"id\":0,\"alias\":\"umbracoMemberPasswordRetrievalAnswer\",\"value\":\"\"},{\"id\":0,\"alias\":\"umbracoMemberPasswordRetrievalQuestion\",\"value\":\"\"},{\"id\":0,\"alias\":\"rootNode\",\"value\":\"umb://document/9023c70aeb2149669f71a4b805a0f1ad\"}],\"name\":\"dsfsdf\",\"contentTypeAlias\":null,\"parentId\":-1,\"action\":\"saveNew\",\"key\":\"5358986b4e20448fa7424759a4e36245\",\"email\":\"...@...se\",\"username\":\"sdfsdf\",\"password\":{\"newPassword\":\"....",\"reset\":true,\"answer\":null},\"memberGroups\":[\"Företagskund\"],\"comments\":\"\",\"isApproved\":true,\"isLockedOut\":false}\r\n------WebKitFormBoundaryt0BjYAAZkFW5wK8D--\r\n"}
```

The server does not return the content-object expected so this is why we get the front end error from the "getAllProps" that is trying to iterate the tabs-property that is not present since an error was returned.

Capture

Possible workarounds at the moment:

- Run the site with debug=true
- Give the User access to the "Settings"-section

I'm a little bit surprised that the debug-flag has any impact on this but it actually seem to have, I've tested this by bumping the clientDependencyVersionNumber and also removing all files in /App_Data/Temp/ClientDependency/* so I can't see that it's related to a cache-issue. When the User have access to the "Settings"-section the debug-mode does not matter, then it works all the time.

### PR DESCRIPTION
**Pull request by enkelmedia**
### Prerequisites
- [x] I have added steps to test this contribution in the description below

https://github.com/umbraco/Umbraco-CMS/issues/8004

### Description
Solves this issue: https://github.com/umbraco/Umbraco-CMS/issues/8004 where a User without access to the settings-section could not create a member since umb-node-preview-directive is setting scope.alias to null.

See the issue for steps to test: https://github.com/umbraco/Umbraco-CMS/issues/8004

I removed the check for whether or not the site is in compilation debug, it's not relevant for this since the a user without access to the settings-section would not be able to access settings when debug=true, I'm checking if the current user have access to the settings-section which feels like the right way to approach this.

### CODE REVIEW

**Comment by nul800sebastiaan** [comment]
Thanks @enkelmedia! There was a small misunderstanding here in that the alias should only be shown when debug is enabled. I've updated this PR to cover all of the cases now. I'll get someone to review it so it will be fixed in the next patch release!

**Comment by enkelmedia** [comment]
Sweet! Thanks for adjusting The PR :) 

Looking forward to it :)

Cheers!

**Comment by nielslyngsoe** [comment]
Thanks @enkelmedia and @nul800sebastiaan, I have tested and it works.
I also took the freedom to change the local variable to provide a clear indication of its intention. I wanted to avoid anyone thinking that this variable provides the property alias. Since I now re-wrote all the code I will ask @nul800sebastiaan to do another approve and merge?

File: src/Umbraco.Web.UI.Client/src/views/components/umb-node-preview.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
         <i ng-if="icon" class="umb-node-preview__icon {{ icon }}" aria-hidden="true"></i>
         <div class="umb-node-preview__content">
 
-            <div class="umb-node-preview__name" ng-attr-title="{{alias}}">{{ name }}</div>
+            <div class="umb-node-preview__name" ng-attr-title="{{::showNodeAlias ? alias : null}}">{{ name }}</div>
```
Comment (ronaldbarendse) [inline src/Umbraco.Web.UI.Client/src/views/components/umb-node-preview.html:6]: This is using double colon/one-time binding syntax now, but doesn't that prevent updating the value after the `userService.getCurrentUser()` promise is fulfilled and asynchronously updates `scope.showNodeAlias`?

**Comment by nul800sebastiaan** [comment]
I realized that we don't need to even get the user data (see if they can access the "settings" section) when `debug="false"` so I've updated the code to avoid that check if it's not needed.

File: src/Umbraco.Web.UI.Client/src/views/components/umb-node-preview.html (hunk 1 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/views/components/umb-node-preview.html:6]: Good point, It did not prevent that, but for safety lets avoid it. A change is coming very soon.

**Comment by nul800sebastiaan** [comment]
Cherry picked for 8.6.2 in 2b4968119ea3380f0aff174807fee1cf79d48693

**Pull request merged**